src/biogeochem/CNCStateUpdate1Mod.F90

```diff
@@ -225,20 +252,28 @@ subroutine CStateUpdate1( num_soilc, filter_soilc, num_soilp, filter_soilp, &
          do j = 1,nlevdecomp
             do fc = 1,num_soilc
                c = filter_soilc(fc)
-               cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) = &
+               if (.not. use_soil_matrixcn) then
+                  cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) = &
                     cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) &
                     - ( cf_soil%decomp_cascade_hr_vr_col(c,j,k) + cf_soil%decomp_cascade_ctransfer_vr_col(c,j,k)) *dt
-            end do
+!                else
+!                  cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_donor_pool(k)) = 0.0_r8
+               end if !use soil_matrix 
+           end do
          end do
       end do
       do k = 1, ndecomp_cascade_transitions
          if ( cascade_receiver_pool(k) /= 0 ) then  ! skip terminal transitions
             do j = 1,nlevdecomp
                do fc = 1,num_soilc
                   c = filter_soilc(fc)
-                  cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_receiver_pool(k)) = &
+                  if (.not. use_soil_matrixcn) then
+                    cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_receiver_pool(k)) = &
                        cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_receiver_pool(k)) &
                        + cf_soil%decomp_cascade_ctransfer_vr_col(c,j,k)*dt
+!                 else 
+!                    cf_soil%decomp_cpools_sourcesink_col(c,j,cascade_receiver_pool(k)) = 0.0_r8
+                  end if !use soil_matrix
```

Comment:
Please change the comment after end if. 
This `end if` is related to line 270: `if (.not. use_soil_matrixcn) then`

I suggest changing it to `!not use_matrixcn`

Reply:
Do you mean change it to !not use_soil_matrixcn ?

Reply:
Looks like this was done.

Reply:
What I meant was to change the comment after `end if` in line 276 to `!not use_matrix`, just to make it easier to follow/read the nested loop. This is not a big deal, so we can leave it as is now.

Reply:
I changed it to "!not use_soil_matrixcn", because it should be related to soil matrix. It should be done.